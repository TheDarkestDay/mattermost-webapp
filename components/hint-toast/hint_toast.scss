@import 'compass/css3';

.hint-toast {
  position: absolute;
  left: 50%;
  z-index: 3;

  margin-top: 16px;
  margin-left: 0;
  margin-right: 0;
  padding-top: 8px;
  padding-bottom: 8px;
  border: 1px solid var(--center-channel-color-16);

  font-size: 14px;
  font-weight: bold;
  color: var(--center-channel-color);

  background-color: var(--center-channel-bg);
  @include border-radius(4px);
  transform: translateX(-50%);
  box-shadow: 0px 6px 14px rgba(0, 0, 0, 0.12);

  .hint-toast__message {
      display: inline-block;
      padding-left: 12px;

      line-height: 20px;
      text-align: center;
      vertical-align: middle;
  }

  .hint-toast__dismiss {
      display: inline-block;
      padding: 0 8px;

      &:hover {
          background-color: var(--center-channel-bg);

          svg {
              fill: var(--center-channel-color-72);
          }
      }

      svg {
          vertical-align: middle;

          fill: var(--center-channel-color-56);
      }

      .close-btn {
          line-height: normal;
      }
  }
}

.hint-toast--expanded {
  left: auto;

  display: flex;
  width: calc(100% - 30px);
  margin: 10px 15px 0px;

  transform: none;

  .hint-toast__dismiss {
    margin-left: auto;
  }
}