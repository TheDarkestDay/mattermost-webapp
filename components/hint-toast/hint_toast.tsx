// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';

import CloseIcon from 'components/widgets/icons/close_icon';

import './hint_toast.scss';

type Props = {
    children: React.ReactNode;
    onDismiss: () => void;
    width: number;
}

const EXPANDED_TOAST_MAX_CONTAINER_WIDTH = 750;

export const HintToast: React.FC<Props> = ({children, onDismiss, width}: Props) => {
    const handleDismiss = () => {
        if (typeof onDismiss === 'function') {
            onDismiss();
        }
    };

    let toastClassName = 'hint-toast';
    if (width < EXPANDED_TOAST_MAX_CONTAINER_WIDTH) {
        toastClassName += ' hint-toast--expanded';
    }

    return (
        <div className={toastClassName}>
            <div
                className='hint-toast__message'
            >
                {children}
            </div>
            <div
                className='hint-toast__dismiss'
                onClick={handleDismiss}
                data-testid='dismissHintToast'
            >
                <CloseIcon
                    className='close-btn'
                    id='dismissHintToast'
                />
            </div>
        </div>
    );
};
