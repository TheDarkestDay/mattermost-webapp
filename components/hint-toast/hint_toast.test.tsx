// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import { shallow } from 'enzyme';

import { HintToast } from './hint_toast';

describe('components/HintToast', () => {
  test('should match snapshot for regular width', () => {
    const wrapper = shallow(<HintToast width={1200} onDismiss={jest.fn()}>A hint</HintToast>);
    expect(wrapper).toMatchSnapshot();
  });

  test('should match snapshot for full width', () => {
    const wrapper = shallow(<HintToast width={400} onDismiss={jest.fn()}>A hint</HintToast>);
    expect(wrapper).toMatchSnapshot();
  });

  test('should fire onDismiss callback', () => {
    const dismissHandler = jest.fn();
    const wrapper = shallow(<HintToast width={1200} onDismiss={dismissHandler}>A hint</HintToast>);

    wrapper.find('.hint-toast__dismiss').simulate('click');
    
    expect(dismissHandler).toHaveBeenCalled();
  });
});
