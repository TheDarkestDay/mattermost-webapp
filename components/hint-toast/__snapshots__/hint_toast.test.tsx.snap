// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/HintToast should match snapshot for full width 1`] = `
<div
  className="hint-toast hint-toast--expanded"
>
  <div
    className="hint-toast__message"
  >
    A hint
  </div>
  <div
    className="hint-toast__dismiss"
    data-testid="dismissHintToast"
    onClick={[Function]}
  >
    <CloseIcon
      className="close-btn"
      id="dismissHintToast"
    />
  </div>
</div>
`;

exports[`components/HintToast should match snapshot for regular width 1`] = `
<div
  className="hint-toast"
>
  <div
    className="hint-toast__message"
  >
    A hint
  </div>
  <div
    className="hint-toast__dismiss"
    data-testid="dismissHintToast"
    onClick={[Function]}
  >
    <CloseIcon
      className="close-btn"
      id="dismissHintToast"
    />
  </div>
</div>
`;
